tools/hula/proxy/src/proxy.rs

```diff
@@ -0,0 +1,222 @@
+use std::{
+    collections::HashMap,
+    io::{BufWriter, Read, Write},
+    mem::size_of,
+    os::unix::{io::AsRawFd, net::UnixStream, prelude::RawFd},
+    ptr::read,
+    slice::from_raw_parts,
+    thread::sleep,
+    time::{Duration, Instant},
+};
+
+use color_eyre::eyre::{bail, Result, WrapErr};
+use epoll::{ControlOptions, Event, Events};
+use hula_types::{Battery, HulaControlFrame, RobotState};
+use log::{debug, error, info, warn};
+use rmp_serde::{encode::write_named, from_slice};
+
+use crate::{
+    idle::{charging_skull, send_idle},
+    listener::HulaListener,
+};
+
+const HULA_SOCKET_PATH: &str = "/tmp/hula";
+const HULA_TIMEOUT: Duration = Duration::from_secs(1);
+const LOLA_SOCKET_PATH: &str = "/tmp/robocup";
+const LOLA_SOCKET_RETRY_COUNT: usize = 60;
+const LOLA_SOCKET_RETRY_INTERVAL: Duration = Duration::from_secs(1);
+
+fn wait_for_lola() -> Result<UnixStream> {
+    for _ in 0..LOLA_SOCKET_RETRY_COUNT {
+        if let Ok(socket) = UnixStream::connect(LOLA_SOCKET_PATH) {
+            return Ok(socket);
+        }
+        info!("Waiting for LoLA socket to become available...");
+        sleep(LOLA_SOCKET_RETRY_INTERVAL);
+    }
+    bail!("stopped after {} retries", LOLA_SOCKET_RETRY_COUNT)
+}
+
+pub struct Proxy {
+    lola: UnixStream,
+    hula: HulaListener,
+    epoll_fd: RawFd,
+}
+
+impl Proxy {
+    pub fn initialize() -> Result<Self> {
+        let lola = wait_for_lola().wrap_err("failed to connect to LoLA")?;
+        let hula = HulaListener::bind(HULA_SOCKET_PATH)
+            .wrap_err_with(|| format!("failed to bind {}", HULA_SOCKET_PATH))?;
+
+        let epoll_fd = epoll::create(false).wrap_err("failed to create epoll file descriptor")?;
+        add_to_epoll(epoll_fd, lola.as_raw_fd())
+            .wrap_err("failed to register LoLA file descriptor in epoll")?;
+        add_to_epoll(epoll_fd, hula.as_raw_fd())
+            .wrap_err("failed to register hula file descriptor in epoll")?;
+
+        Ok(Self {
+            lola,
+            hula,
+            epoll_fd,
+        })
+    }
+
+    pub fn run(mut self) -> Result<()> {
+        let proxy_start = Instant::now();
+        let mut connections = HashMap::new();
+        let mut events = [Event::new(Events::empty(), 0); 16];
+        let mut writer = BufWriter::with_capacity(786, self.lola.try_clone()?);
+        let mut battery = None;
+
+        debug!("Entering epoll loop...");
+        loop {
+            let epoll_timeout = -1;
+            let num_events = epoll::wait(self.epoll_fd, epoll_timeout, &mut events)
```

Comment:
Please don't abbreviate 

```suggestion
            let number_of_events = epoll::wait(self.epoll_file_descriptor, epoll_timeout, &mut events)
```

Please search and replace everywhere. Thanks.

Reply:
boah... alter fine.. ... done

Reply:
You fixed `number_of_events` but we still have a lot of `fd` occurrences.

Reply:
I refuse to replace `fd`.  The code becomes extremely ugly and not readable anymore. Lines break, and the standard library itself uses `as_raw_fd()` this is very compact and repeatedly readable.

Reply:
The standard library sadly is no benchmark. It has very bad naming in some places. For me it would be more important to have a good name without abbreviations than have pretty formatted code. Good names are also important for readability. I will create a PR on my own if you don't want to do it. Please resolve or write it out.